spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -601,7 +601,7 @@ private void invokeBatchListener(final ConsumerRecords<K, V> records) {
 					}
 				}
 				catch (Exception e) {
-					if (this.containerProperties.isAckOnError() && !this.autoCommit) {
+					if (this.containerProperties.isAckOnError() && !this.autoCommit && !this.isAnyManualAck) {
```

Comment:
Since we can't commit error in case of any manual mode, it sounds like better to move that logic to the `containerProperties.isAckOnError()` directly:
```
public boolean isAckOnError() {
	return this.ackOnError && 
			!(AckMode.MANUAL_IMMEDIATE.equals(this.ackMode) || AckMode.MANUAL.equals(this.ackMode));
}
```